### ISSUE TITLE
Assess OmniPath, create secondary adapter

### ISSUE THREAD

**Issue by slobentanzer** [biocypher/DFG528753569#1]
For initial assessment of the best strategy of migrating OmniPath to BioCypher, we will build a pipeline with secondary adapter and create a representative schema of the OmniPath KG to be developed. Dev repository is https://github.com/biocypher/omnipath.

**Comment by ecarrenolozano** [comment]
In OmniPath there are 5 databases: 

- network (interactions)
- enzyme-substrate interactions (enz_sub or ptms)
- protein complexes (complexes)
- molecular entity annotations (annotations)
- intercellular communication roles (intercell).

We can access each one by using the [OmniPath Python client](https://omnipath.readthedocs.io/en/latest/index.html). However, there is a single dataframe called "omnipath" (see this [link](https://omnipath.readthedocs.io/en/latest/api/omnipath.interactions.OmniPath.html#omnipath.interactions.OmniPath)). 

**Q1.:** When we refer to OmniPath for the secondary adapter, are we talking about the complete set (5 databases) or just this dataset?
**Q2.:** Where can I consult the metadata for each database? For instance, number of tables, number of columns, numbers of rows.

**Comment by slobentanzer** [comment]
Q1: While this is more complicated, as first target I would aim at the first component, the interactions, as a prototype. This seems to be the core of OmniPath, and definitely the most-used part of it. We can always extend later, if this is desired.

Q1-extended: I would propose (after the initial prototype is there) to not blindly reproduce OmniPath, but rather look at the downstream use cases and engineer a solutions that brings most flexibility in the parts of OmniPath that are actually regularly used, while ignoring the parts that are not.

Q2: I am not sure (also not sure there is such a thing). If very important, you can open an issue in pypath.

**Comment by ecarrenolozano** [comment]
> Q1: While this is more complicated, as first target I would aim at the first component, the interactions, as a prototype. This seems to be the core of OmniPath, and definitely the most-used part of it. We can always extend later, if this is desired.
> 
> Q1-extended: I would propose (after the initial prototype is there) to not blindly reproduce OmniPath, but rather look at the downstream use cases and engineer a solution that brings most flexibility in the parts of OmniPath that are actually regularly used, while ignoring the parts that are not.
> 
 I agree, so our first focus will be working with the networks (interactions) data.

> Q2: I am not sure (also not sure there is such a thing). If very important, you can open an issue in pypath.

This is important to check the consistency of the data, if the data we downloaded is what we expected.

**Comment by slobentanzer** [comment]
You can go ahead with the networks data that you sent to me, it looks like it should. I would not worry too much about anything beyond that before we have the prototype. Once you have it, you can - for instance - implement a check to compare whether randomly sampled interactions from the OmniPath API are also present in your graph. I think checking the data is more robust than checking metadata, anyways.

**Comment by ecarrenolozano** [comment]
Development closed in https://github.com/biocypher/omnipath-secondary-adapter
